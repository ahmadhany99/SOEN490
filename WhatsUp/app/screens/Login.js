/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { useState } from "react";
import { StyleSheet, Text, View, Image, Alert } from "react-native";
import AppButton from "../components/AppButton";
import AppTextInput from "../components/AppTextInput";
import Links from "../components/Links";
import Screen from "../components/Screen";
import TitleHeaders from "../components/TitleHeaders";
import logo from "../Images/w3.png";
import { auth, db } from "../firebase";
import { collection, getDocs, query, where } from "firebase/firestore";
import { signInWithEmailAndPassword } from "firebase/auth";
import { useNavigation } from "@react-navigation/native";
import colors from "../config/colors";
import { SimpleLineIcons } from "@expo/vector-icons";

export default function Login() {
  const navigation = useNavigation();
  const [email, setEmail] = useState("jasmine@organizer.com");
  const [password, setPassword] = useState("capstone123");
  const [isOrganizer] = useState(false);
  const handleLogin = async () => {
    await signInWithEmailAndPassword(auth, email, password)
      .then(async () => {
        const q = query(collection(db, "users"), where("email", "==", email));
        const querySnapshot = await getDocs(q);
        if (querySnapshot != null) {
          querySnapshot.forEach((doc) => {
            if (isOrganizer == doc.data().isOrganizer) {
              navigation.navigate("Attendee");
              setPassword("");
            } else {
              navigation.navigate("Organizer");
              setPassword("");
            }
          });
        }
      })
      .catch(() => {
        Alert.alert("Try again", "Invalid email or password.");
      });
  };
  const SignupPressed = async () => {
    navigation.navigate("SignUp");
  };

  const ResetPassword = () => {
    navigation.navigate("ResetPassword");
  };
  return (
    <Screen
      resizeMode='cover'
      backgroundImage='https://cdn.dribbble.com/users/479289/screenshots/4521207/media/c3e3bd246bbff91101a54e69daa8b1f0.gif'
      style={{ paddingHorizontal: "10%" }}
    >
      <Image
        source={logo}
        style={{ marginTop: 18, width: 166, height: 212, alignSelf: "center" }}
      />
      <TitleHeaders
        style={{ fontSize: 38, marginTop: 20, marginBottom: 5 }}
        title='Login'
        isTitle={true}
      />
      <AppTextInput
        placeholder='Email'
        keyboardType='email-address'
        onChangeText={(text) => setEmail(text.toLowerCase())}
        value={email}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <AppTextInput
        placeholder='Password'
        secureTextEntry
        onChangeText={(text) => setPassword(text)}
        value={password}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <View style={styles.organizertwo}>
        <Links
          style={styles.link}
          link='Forgot Password?'
          onPress={ResetPassword}
        />
      </View>
      <AppButton title='Login' onPress={handleLogin} />
      <View style={{ flexDirection: "row", justifyContent: "center" }}>
        <Text style={styles.text}>Don't have an account? </Text>
        <Links style={styles.link} link='Sign up' onPress={SignupPressed} />
      </View>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          paddingVertical: 15,
        }}
      >
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
        <View>
          <Text
            style={{
              width: 105,
              textAlign: "center",
              color: "darkgray",
              fontSize: 14,
            }}
          >
            Or login with
          </Text>
        </View>
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
      </View>
      <View style={styles.rowContainer}>
        <SimpleLineIcons
          name='social-facebook'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            alignSelf: "center",
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='facebook'
        />
        <SimpleLineIcons
          name='social-twitter'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingHorizontal: 40,
            alignSelf: "center",
          }}
          onPress={() => console.log("Facebook")}
          testID='twitter'
        />
        <SimpleLineIcons
          name='social-google'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='google'
        />
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  organizer: {
    alignItems: "flex-start",
    width: "50%",
  },
  organizertwo: {
    alignItems: "flex-end",
    width: "98%",
  },
  rowContainer: {
    flexDirection: "row",
    alignSelf: "center",
  },
  roundshape: {
    backgroundColor: "grey",
    height: 65,
    width: 65,
    justifyContent: "center",
    alignSelf: "center",
    alignContent: "center",
    borderRadius: 45,
  },
  text: {
    color: colors.darkerGrey,
    fontSize: 16,
  },
  text2: {
    fontSize: 16,
    color: colors.darkGrey,
  },
});
