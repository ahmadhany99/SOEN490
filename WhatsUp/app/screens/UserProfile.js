/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import { View, Image, StyleSheet, ScrollView, Alert } from "react-native";
import Screen from "../components/Screen";
import ScreenTitle from "../components/ScreenTitle";
import { Ionicons } from "@expo/vector-icons";
import logo from "../Images/w3.png";
import ImgOrgBottom from "../components/ImgOrgBottom";
import ScreenSubtitle from "../components/ScreenSubtitle";
import Links from "../components/Links";
import colors from "../config/colors";
import AppModal from "../components/AppModal";
import BackBtn from "../components/BackBtn";
import AppTextInput from "../components/AppTextInput";
import AppButton from "../components/AppButton";
import { useAuthState } from "react-firebase-hooks/auth";
import { auth, db } from "../firebase";
import { sendPasswordResetEmail, updateEmail } from "firebase/auth";
import {
  collection,
  getDocs,
  query,
  where,
  doc,
  updateDoc,
} from "firebase/firestore";
import { fetchSignInMethodsForEmail } from "firebase/auth";

function UserProfile() {
  const [modalVisibleName, setModalVisibleName] = useState(false);
  const [modalVisibleEmail, setModalVisibleEmail] = useState(false);
  const [newName, setNewName] = useState("");
  const [newEmail, setNewEmail] = useState("");
  const [docKey, setDocKey] = useState("");
  const [userName, setUserName] = useState("");
  const [user] = useAuthState(auth);

  const getName = async () => {
    const q = query(collection(db, "users"), where("email", "==", user.email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot != null) {
      querySnapshot.forEach((doc) => {
        setUserName(doc.data().firstName + " " + doc.data().lastName);
      });
    }
  };

  const getKey = async () => {
    const q = query(collection(db, "users"), where("email", "==", user.email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot != null) {
      querySnapshot.forEach((doc) => {
        setDocKey(doc.id);
      });
    }
  };

  getName();
  getKey();

  const updateFirestoreEmail = async () => {
  const docRef = doc(db, "users", docKey);
  const data = {
    email: newEmail,
  };
  await updateDoc(docRef, data)
    .then(() => {
    })
    .catch((error) => {
      console.log(error);
    });
  }

  const updateEmailForCurrentUser = () => {
    updateEmail(user, newEmail)
      .then(() => {
        // Email updated!
        // ...
      })
      .catch((error) => {
        // An error occurred
        // ...
      });
  };

  const sendResetEmail = () => {
    sendPasswordResetEmail(auth, user.email)
      .then(() => {
        Alert.alert(
          "Email sent sucessfully.",
          "Check your emails to reset the password."
        );
      })
      .catch(() => {});
  };

  const checkIfEmailExists = () => {
    fetchSignInMethodsForEmail(auth, user.email)
      .then((result) => {
        if (result === undefined || result.length == 0) {
          Alert.alert("We couldn't find an account with that email address.");
        } else {
          //If email exists, send the link
          sendResetEmail();
        }
      })
      .catch((error) => {
        console.log(error.message);
        Alert.alert(error.message);
      });
  };

  const checkIfEmailAlreadyExists = () => {
    fetchSignInMethodsForEmail(auth, newEmail)
      .then((result) => {
        if (result === undefined || result.length == 0) {
          Alert.alert("Email was successfully updated.");
          updateEmailForCurrentUser(newEmail);
        } else {
          Alert.alert("The email already exists.");
        }
      })
      .catch((error) => {
        console.log(error.message);
        Alert.alert(error.message);
      });
  };

  return (
    <Screen style={{ padding: 20, backgroundColor: "#F5F5F5" }}>
      <View style={{ left: "1.5%", marginTop: "5%" }}>
        <ScreenTitle title="Profile" style={{ marginBottom: "2%" }} />

        <Image
          source={logo}
          style={{
            width: 166,
            height: 212,
            alignSelf: "center",
            marginTop: "5%",
          }}
        />
      </View>
      <View style={{ flexDirection: "row" }}>
        <ScreenTitle
          title="Name"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "10%" }}
        />
        {/* code if we ever want to edit the name: */}
        {/* <Ionicons
          name="pencil"
          onPress={() => setModalVisibleName(true)}
          size={20}
          color={colors.primary}
          style={{
            marginBottom: "2%",
            marginTop: "10%",
            alignSelf: "flex-end",
            marginLeft: "auto",
          }}
        /> */}
      </View>
      <View>
        <ScreenSubtitle
          subtitle={userName}
          style={{
            marginBottom: "2%",
            marginTop: "0.5%",
            marginLeft: "1%",
            fontSize: 16,
          }}
        />
      </View>
      <View style={{ flexDirection: "row" }}>
        <ScreenTitle
          title="Email"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "5%" }}
        />
        <Ionicons
          name="pencil"
          onPress={() => setModalVisibleEmail(true)}
          size={20}
          color={colors.primary}
          style={{
            marginBottom: "2%",
            marginTop: "5%",
            alignSelf: "flex-end",
            marginLeft: "auto",
          }}
        />
      </View>
      <View>
        <ScreenSubtitle
          subtitle={user.email}
          style={{
            marginBottom: "2%",
            marginTop: "0.5%",
            marginLeft: "1%",
            fontSize: 16,
          }}
        />
        <ScreenTitle
          title="Password"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "5%" }}
        />
        <Links
          style={{ marginLeft: "1%", marginTop: "0.5%", fontSize: 16
           }}
          link="Change Password"
          onPress={checkIfEmailExists}
        />
      </View>
      <ImgOrgBottom resizeMode="contain" />
      <AppModal
        animationType="slide"
        transparent={true}
        visible={modalVisibleName}
        onRequestClose={() => {
          setModalVisibleName(!modalVisible);
        }}
      >
        <View style={styles.modalView}>
          <BackBtn
            style={styles.backModal}
            onPress={() => setModalVisibleName(!modalVisibleName)}
          />
          <View style={styles.inputView}>
            <ScrollView
              keyboardDismissMode="interactive"
              style={{ width: "100%" }}
            >
              <ScreenTitle
                style={{
                  alignSelf: "center",
                  fontSize: 22,
                  marginBottom: "5%",
                }}
                title={"Enter New Name:"}
              />
              <AppTextInput
                placeholder="Name"
                onChangeText={(currentName) => setNewName(currentName)}
              />
              <AppButton
                title="Submit"
                style={{ marginTop: 0 }}
                onPress={() => {
                  setModalVisibleName(!modalVisibleName);
                }}
              />
            </ScrollView>
          </View>
        </View>
      </AppModal>
      <AppModal
        animationType="slide"
        transparent={true}
        visible={modalVisibleEmail}
        onRequestClose={() => {
          setModalVisibleName(!modalVisibleName);
        }}
      >
        <View style={styles.modalView}>
          <BackBtn
            style={styles.backModal}
            onPress={() => setModalVisibleEmail(!modalVisibleEmail)}
          />
          <View style={styles.inputView}>
            <ScrollView
              keyboardDismissMode="interactive"
              style={{ width: "100%" }}
            >
              <ScreenTitle
                style={{
                  alignSelf: "center",
                  fontSize: 22,
                  marginBottom: "5%",
                }}
                title={"Enter New Email:"}
              />
              <AppTextInput
                placeholder="Email"
                onChangeText={(newEmail) => {
                  setNewEmail(newEmail);
                }}
              />
              <AppButton
                title="Submit"
                style={{ marginTop: 0 }}
                onPress={() => {
                  setModalVisibleEmail(!modalVisibleEmail);
                  checkIfEmailAlreadyExists();
                  updateFirestoreEmail(newEmail);
                }}
              />
            </ScrollView>
          </View>
        </View>
      </AppModal>
    </Screen>
  );
}

const styles = StyleSheet.create({
  modalView: {
    margin: 270,
    backgroundColor: "white",
    borderRadius: 35,
    padding: 20,
    paddingTop: 20,
    width: "86%",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  inputView: {
    marginTop: 8,
    borderColor: colors.lightGrey,
    borderRadius: 7,
    width: "90%",
    alignSelf: "center",
    paddingTop: 20,
  },
  backModal: {
    backgroundColor: "black",
  },
});

export default UserProfile;
