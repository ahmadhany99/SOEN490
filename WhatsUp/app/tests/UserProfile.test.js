import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import { Alert } from "react-native";
import UserProfile from "../screens/UserProfile";
import Login from "../screens/Login";

jest.useFakeTimers();

it("Renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <UserProfile />
    </NavigationContainer>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Successful log out and go back to Login page", async () => {
  render(
    <NavigationContainer>
      <UserProfile />
    </NavigationContainer>
  );

  await waitFor(() => fireEvent.press(screen.getByText("Log out")));
  const ls = (
    <NavigationContainer>
      <Login />
    </NavigationContainer>
  );
  render(ls);
  expect(screen.getByText("Please enter your details"));
});


