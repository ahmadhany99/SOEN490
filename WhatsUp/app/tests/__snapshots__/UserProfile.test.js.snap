// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login sucessfully 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "backgroundColor": "#F5F5F5",
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "left": "1.5%",
        "marginTop": "5%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "sans-serif",
            "marginLeft": "1%",
          },
          Object {
            "marginBottom": "2%",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Profile
    </Text>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 212,
          "marginTop": "5%",
          "width": 166,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "10%",
        }
      }
    >
      Name
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Email
    </Text>
    <Text />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Password
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "marginLeft": "1%",
            "marginTop": "0.5%",
          }
        }
      >
        Change Password
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text />
  </View>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "bottom": 0,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "bottom": 0,
          "height": 172.91414752116083,
          "position": "absolute",
          "resizeMode": "stretch",
          "width": 750,
        },
      ]
    }
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;

exports[`Renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "backgroundColor": "#F5F5F5",
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "left": "1.5%",
        "marginTop": "5%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "sans-serif",
            "marginLeft": "1%",
          },
          Object {
            "marginBottom": "2%",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Profile
    </Text>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 212,
          "marginTop": "5%",
          "width": 166,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "10%",
        }
      }
    >
      Name
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Email
    </Text>
    <Text />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Password
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "marginLeft": "1%",
            "marginTop": "0.5%",
          }
        }
      >
        Change Password
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text />
  </View>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "bottom": 0,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "bottom": 0,
          "height": 172.91414752116083,
          "position": "absolute",
          "resizeMode": "stretch",
          "width": 750,
        },
      ]
    }
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;

exports[`Renders correctly user profile page 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "backgroundColor": "#F5F5F5",
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "left": "1.5%",
        "marginTop": "5%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "sans-serif",
            "marginLeft": "1%",
          },
          Object {
            "marginBottom": "2%",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Profile
    </Text>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 212,
          "marginTop": "5%",
          "width": 166,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "10%",
        }
      }
    >
      Name
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Email
    </Text>
    <Text />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "0.5%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": "2%",
          "marginLeft": "1%",
          "marginTop": "5%",
        }
      }
    >
      Password
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "marginLeft": "1%",
            "marginTop": "0.5%",
          }
        }
      >
        Change Password
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text />
  </View>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "bottom": 0,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "bottom": 0,
          "height": 172.91414752116083,
          "position": "absolute",
          "resizeMode": "stretch",
          "width": 750,
        },
      ]
    }
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;
